mmv1/compile/openapi_parser.rb

```diff
@@ -0,0 +1,153 @@
+require 'openapi_parser'
+
+def writeObject(name, obj, type, url_param)
+  res = nil
+  case type
+  when "string"
+    res = Api::Type::String.new(name)
+  when "integer"
+    res = Api::Type::Integer.new(name)
+  when "number"
+    res = Api::Type::Double.new(name)
+  when "boolean"
+    res = Api::Type::Boolean.new()
+    res.instance_variable_set(:@name, name)
+  when "object"
+    if name == "labels"
+      # standard labels field handling
+      res = Api::Type::KeyValuePairs.new()
+    else
+      res = Api::Type::NestedObject.new()
+      req = obj.required || []
```

Comment:
nit: consider longer variable names for clarity. I keep reading "req" as "request" personally! Totally stylistic though, and this guidance is contrary to at least official Go guidance (idk what would be recommended in Ruby), so up to you.